pkg/codegen/importer/hcl2.go

```diff
@@ -365,11 +366,222 @@ func generatePropertyValue(property *schema.Property, value resource.PropertyVal
 	return generateValue(property.Type, value)
 }
 
+// valueStructurallyTypedAs returns true if the given value is structurally typed as the given schema type.
+func valueStructurallyTypedAs(value resource.PropertyValue, schemaType schema.Type) bool {
+	switch {
+	case value.IsObject():
+		switch arg := schemaType.(type) {
+		case *schema.ObjectType:
+			schemaProperties := make(map[string]schema.Type)
```

Comment:
This map is built into `schema.ObjectType`. The API is [here](https://github.com/pulumi/pulumi/blob/16260035bb7c32e5883bbe14061b2a19cf6a45c3/pkg/codegen/schema/schema.go#L256).